flows/aws/interactive-workflow/marvin_extension.py

```diff
@@ -0,0 +1,121 @@
+import marvin
+from prefect import flow, get_run_logger, pause_flow_run, task
+from prefect.blocks.system import JSON, Secret
+from prefect.input import RunInput
+from prefect.variables import Variable
+from prefect_aws.s3 import S3Bucket
+from pydantic import BaseModel, Field, constr
+
+DEFAULT_EXTRACT_QUERY = "Group by location and count the number of users in each location."  # Create a table of a users name, location, coordinates, and continent the user is located
+GENERATE_SUGGESTED_FILE_NAME = (
+    "Output a 10-letter single word that describes the user's query: "
+)
+
+
+class userApprovalAndFileName(RunInput):
+    # file_name: constr(pattern=r"^[a-zA-Z]+$", max_length=10)
+    file_name: str
+    approve: bool = Field(description="Would you like to approve?")
+
+
+class InputQuery(RunInput):
+    input_instructions: str
+
+
+class generatedFileName(BaseModel):
+    fixed_length_string: constr(pattern=r"^[a-zA-Z]+$", min_length=10, max_length=10)
+
+
+@flow(name="Extract User Insights")
+def extract_information():
+    secret_block = Secret.load("openai-creds-interactive-workflows")
+    marvin.settings.openai.api_key = secret_block.get()
+    logger = get_run_logger()
+
+    features = JSON.load("all-users-json").value
```

Comment:
Why is the data getting passed as a `JSON` block?

Reply:
I wanted to use the JSON block to showcase some 'data lineage' on the query results. Plus there are events thrown each time this block is used/updated, so any trigger based work can build off of that

Reply:
I think it might make more sense to enjoy a custom event in that case but for the purposes of this demo I don't think that's necessary.